### PR TITLE
CB-11799 Add API to check users and groups in ipa server

### PR DESCRIPTION
**Pull request by horadla23**
See detailed description in the commit message.

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/client/FreeipaClientTestService.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+        try {
+            User user = freeIpaClient.userShow(name);
+            LOGGER.info("Groups: {}", user.getMemberOfGroup());
+            LOGGER.info("Success: {}", user);
+        } catch (Exception e) {
+            LOGGER.error("Error showing user {}", name, e);
+            return "FAILED TO SHOW USER";
+        }
+        return "END";
```
Comment (horadla23) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/client/FreeipaClientTestService.java:48]: I want to declare, I just moved this logic into the service, but I do not wanted to touch it, even if I am not agree with the desing of it

Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/client/FreeipaClientTestService.java:48]: it wasn't designed :D

Comment (horadla23) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/client/FreeipaClientTestService.java:48]: yes, it definitely seems it is not created for actual customer use cases :D

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/ClientTestV1Endpoint.java [hunk 2]
```diff
@@ -17,9 +22,29 @@
...
     @Path("{id}/{name}")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "Retrieves user information", produces = MediaType.APPLICATION_JSON, nickname = "userShowV1")
     String userShow(@PathParam("id") Long id, @PathParam("name") String name);
+
+    @POST
+    @Path("check_users")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Checks if users exists in FreeIPA of the environment", produces = MediaType.APPLICATION_JSON, nickname = "checkUsers")
+    Boolean checkUsers(@Valid CheckUsersRequest checkUsersRequest);
```
Comment (lacikaaa) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/ClientTestV1Endpoint.java:36]: are you sure a boolean will be sufficient here later? I would return a response object which could be enhanced later if needed

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckGroupsRequest.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel("CheckGroupsRequest")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CheckGroupsRequest extends ClientTestBase {
+
+    @NotNull
```
Comment (lacikaaa) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckGroupsRequest.java:17]: `@NotNull` is included in `@NotEmpty`, isn't it?

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckUsersInGroupRequest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@ApiModel("CheckUsersInGroupRequest")
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class CheckUsersInGroupRequest extends ClientTestBase {
+
+    @NotNull
+    @NotEmpty
+    @ApiModelProperty(value = "Users to check for", required = true)
+    private Set<String> users;
+
+    @NotNull
```
Comment (lacikaaa) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckUsersInGroupRequest.java:22]: `@NotEmpty` would be better if empty string is invalid

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/client/FreeipaClientTestService.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    public Boolean checkUsers(String environmentCrn, Set<String> requestedUsers) {
+        FreeIpaClient freeIpaClient = getClientByEnvironmentCrn(environmentCrn);
+        try {
+            LOGGER.info("Checking for users [{}] in environment {}", Joiner.on(",").join(requestedUsers), environmentCrn);
+            Set<String> freeipaUsers = freeIpaClient.userFindAll().stream().map(User::getUid).collect(Collectors.toSet());
+            LOGGER.debug("Users in freeipa: [{}]", Joiner.on(",").join(freeipaUsers));
+            return freeipaUsers.containsAll(requestedUsers);
+        } catch (FreeIpaClientException e) {
+            LOGGER.error("Find user FreeIPA call failed!", e);
+            throw new InternalServerErrorException("Find user FreeIPA call failed!");
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/client/FreeipaClientTestService.java:60]: you could use `FreeIpaClientExceptionWrapper`

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/ClientTestV1Endpoint.java (hunk 2 above)
Comment (horadla23) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/ClientTestV1Endpoint.java:36]: makes sense, i wasn't sure if i need that or not, but maybe it is worth to use response object

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckGroupsRequest.java (hunk 3 above)
Comment (horadla23) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckGroupsRequest.java:17]: yes you're right

File: freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckUsersInGroupRequest.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+import javax.validation.constraints.NotEmpty;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.sequenceiq.service.api.doc.ModelDescriptions;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel("CheckUsersInGroupRequest")
```
Comment (lacikaaa) [inline freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckUsersInGroupRequest.java:13]: please add `v1` to the name of the requests as we do it with the others

Comment (horadla23) [reply freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/test/model/CheckUsersInGroupRequest.java]: done

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**